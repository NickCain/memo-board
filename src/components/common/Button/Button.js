import PropTypes from 'prop-types';
import React, { Component } from 'react';
import injectTapEventPlugin from 'react-tap-event-plugin';

import FlatButton from 'material-ui/FlatButton';

import './Button.css';
class Button extends Component {
  render() {
    const { text, onClick } = this.props;
    return (
      <FlatButton className="Button" primary onClick={onClick}>
        {text}
      </FlatButton>
    );
  }
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func
};

injectTapEventPlugin();

export default Button;
