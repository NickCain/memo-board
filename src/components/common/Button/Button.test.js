import { mount, shallow } from 'enzyme';
import React from 'react';
import Button from './Button.js'

let wrapper;
let shallowWrapper;

const mockData = {
  handleOnClick: jest.fn()
};

describe('Button - Component', () => {
  describe('Button renders', () => {
    beforeEach(() => {
      wrapper = mount(
        <Button text="Test" onClick={mockData.handleOnClick}/>
      );
      shallowWrapper = shallow(
        <Button text="Test" onClick={mockData.handleOnClick}/>
      );
    });

    it('should render the Button component', () => {
      expect(shallowWrapper).toMatchSnapshot();
      expect(wrapper.length).toBeGreaterThan(0);
    });

    it('should render text passed as a prop', () => {
      expect(wrapper.text()).toContain('Test');
    });

    it('should fire the cancel function when clicked to remove a meal for All passengers', () => {
      wrapper.find('.Button').simulate('click');
      expect(mockData.handleOnClick).toHaveBeenCalled();
    });
  });
});
