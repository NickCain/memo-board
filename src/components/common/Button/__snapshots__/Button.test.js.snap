// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button - Component Button renders should render the Button component 1`] = `undefined`;
